sensor/common/clusterentities/store_ips.go

```diff
@@ -0,0 +1,217 @@
+package clusterentities
+
+import (
+	"fmt"
+
+	"github.com/stackrox/rox/pkg/net"
+	"github.com/stackrox/rox/pkg/networkgraph"
+	"github.com/stackrox/rox/pkg/sync"
+	"golang.org/x/exp/maps"
+)
+
+type ipsStore struct {
+	mutex sync.RWMutex
+	// entitiesMemorySize defines how many ticks old endpoint data should be remembered after removal request
+	// Set to 0 to disable memory
+	entitiesMemorySize uint16
+
+	// ipMap maps ip addresses to sets of deployment ids this IP is associated with.
+	ipMap map[net.IPAddress]map[string]struct{}
+	// reverseIpMap maps deployment ids to sets of IP addresses associated with this deployment.
+	reverseIPMap map[string]map[net.IPAddress]struct{}
+	// historicalIPs is mimicking ipMap: IP Address -> deploymentID -> historyStatus
+	historicalIPs map[net.IPAddress]map[string]*entityStatus
+}
+
+func newIPsStoreWithMemory(numTicks uint16) *ipsStore {
+	store := &ipsStore{entitiesMemorySize: numTicks}
+	store.mutex.Lock()
+	defer store.mutex.Unlock()
+	store.initMapsNoLock()
+	return store
+}
+
+func (e *ipsStore) initMapsNoLock() {
+	e.ipMap = make(map[net.IPAddress]map[string]struct{})
+	e.reverseIPMap = make(map[string]map[net.IPAddress]struct{})
+	e.historicalIPs = make(map[net.IPAddress]map[string]*entityStatus)
+}
+
+func (e *ipsStore) resetMaps() {
+	e.mutex.Lock()
+	defer e.mutex.Unlock()
+	// Maps holding historical data must not be wiped on reset! Instead, all entities must be marked as historical.
+	// Must be called before the respective source maps are wiped!
+	// Performance optimization: no need to handle history if history is disabled
+	if e.entitiesMemorySize == 0 {
+		e.initMapsNoLock()
+		return
+	}
+	e.markAllIPsHistorical()
+
+	e.ipMap = make(map[net.IPAddress]map[string]struct{})
+	e.reverseIPMap = make(map[string]map[net.IPAddress]struct{})
+	e.updateMetricsNoLock()
+}
+
+func (e *ipsStore) updateMetricsNoLock() {
+}
```

Comment:
Same as above.

Reply:
I actually plan to add some metrics for this... let's see if that is possible in this PR.